src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -229,6 +229,16 @@ else if (!isCharacterValidAfterGenericEnd(charAfter)) {
         }
     }
 
+    /**
+     * Checks if generic is before constructor invocation.
+     * @param ast ast
+     * @return true if generic before a constructor invocation
+     */
+    private static boolean isGenericBeforeCtor(DetailAST ast) {
+        return ast.getParent().getParent().getType() == TokenTypes.LITERAL_NEW
+                && ast.getParent().getNextSibling().getType() == TokenTypes.IDENT;
```

Comment:
When is it not identifier?
And does this cover things like `new OuterClass.InnerClass()` or `new package.Class()`.

Reply:
It is not identifier in cases like new ArrayList<String>

Reply:
Done
I added test cases and code changes to detect static classes and package